src/Physics/Common/XSecScaleI.h

```diff
@@ -0,0 +1,43 @@
+//____________________________________________________________________________
+/*!
+
+\class    genie::XSecScaleI
+
+\brief    This class is responsible to compute a scaling factor for the XSec
+
+\author   Code contributed by J.Tena Vidal and M.Roda
+
+\created  June, 2020
+
+\cpright  Copyright (c) 2003-2020, The GENIE Collaboration
+          For the full text of the license visit http://copyright.genie-mc.org
+*/
+//____________________________________________________________________________
+
+#ifndef _XSEC_SCALE_I_H_
+#define _XSEC_SCALE_I_H_
+
+#include "Framework/Algorithm/Algorithm.h"
+#include "Framework/Interaction/Interaction.h"
+
+namespace genie {
+  
+  class XSecScaleI: public Algorithm {
+    
+  public:    
+    // This function returns the scaling value for a given interaction:
+    virtual double GetScaling( const Interaction & ) const = 0 ; 
+
+  protected:
+    XSecScaleI( string name, string config = "Default" );
+    virtual ~XSecScaleI();
```

Comment:
The destructor protected is almost always an error. Who taught you this monstrosity?